### PR TITLE
Update the manifest used to retrieve platform dependencies and add unit tests

### LINKED ISSUE #27634: [Bindgen] Improve the unit tests inside the bindgen module
**Description:**
The unit tests inside the `bindgen` module could be improved to cover all possible binding generations while avoiding scenarios using the standard library [ballerina/jarrays](https://github.com/ballerina-platform/module-ballerina-java.arrays) (these could be covered inside the `bindgen` tests that are to be added inside the [ballerina-distribution](https://github.com/ballerina-platform/ballerina-distribution) [1]).

Cover the following command options:

**`-cp|--classpath`**

- [ ] Test `-cp` and `--classpath` alternatively
- [x] Incorrect classpath (non-existent jar, a different file or a non existent folder) -> error message
- [ ] Single jar (relative path, absolute path) -> correctly added to classpath
- [ ] Directory (relative path, absolute path) -> jars inside correctly added to the classpath
- [ ] Use multiple comma-separated classpaths (2 jars, 2 directories, 1 jar 1 directory) -> correctly add all to the classpath

**`-mvn|--maven`**
- [ ] Test `-mvn` and `--maven` alternatively
- [x] Incorrect maven library (non-existent jar) -> error message
- [ ] Correct maven library -> jar correctly resolved into `target/platform-libs`
- [ ] Both `classpath` and `maven` options specified -> correctly add them all to the classpath

**`-o|--output`**
- [ ] Test `-o` and `--output` alternatively
- [x] Incorrect output path provided -> error message
- [ ] Relative output path provided -> determine the output path correctly
- [ ] Absolute output path provided -> determine the output path correctly
- [ ] Path resides inside a project -> identify correctly (should resolve existing dependencies and update the `Ballerina.toml` file)
- [ ] Path doesn't reside inside a project -> generate bindings in the directory specified (`target/platform-libs` created here)
- [ ] No output provided -> generate bindings in the user directory (`target/platform-libs` created here)

**`-m|--modules`**
- [ ] Within a project -> generate bindings in separate modules
- [ ] Not within a project -> error message
- [ ] Both output and modules flags provided -> error message

**`--public`**
- [ ] With the flag for an `--output` path -> public bindings
- [ ] Without the flag for an `--output` path -> module private bindings
- [ ] With the flag for no output path -> public bindings
- [ ] Without the flag for no output path -> module private bindings

**`<class-name>`**
- [ ] Incorrect class name -> error message
- [ ] Correct class name -> binding generated
- [ ] No class name -> error message

**Other scenarios:**
- [x] Test if existing path platform libraries and maven platform libraries are resolved by the bindgen tool -> should be added to the classpath

[1] https://github.com/ballerina-platform/ballerina-lang/issues/25600 

**Steps to reproduce:**

**Affected Versions:**
1.2.x and SLP

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### LINKED ISSUE #28550: Update the manifest class used for resolving platform dependencies
**Description:**
In the Bindgen tool, the model that is being used to retrieve the platform dependencies is not the latest one. This needs to be updated to use the latest APIs exposed through the Project API changes.

https://github.com/ballerina-platform/ballerina-lang/blob/9e9cc3a7dfd69633a52b0b31b33892a2e95e0775/misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/command/BindingsGenerator.java#L144

**Steps to reproduce:**

**Affected Versions:**
SLP Alpha 1

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by IrushiL**
## Purpose

* Fixes #28550
* Partially Fixes #27634
* Models the bindgen options as a separate unit.

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/command/BindingsGenerator.java [hunk 1]
```diff
@@ -141,16 +146,36 @@ void generateJavaBindings() throws BindgenException {
...
-            if (manifest != null) {
-                List<Library> platformLibraries = manifest.getPlatform().getLibraries();
-                if (platformLibraries != null) {
-                    for (Library library : platformLibraries) {
-                        if (library.path != null) {
-                            classPaths.add(Paths.get(projectRoot.toString(), library.path).toString());
-                        } else if (library.groupId != null && library.artifactId != null && library.version != null) {
-                            new BindgenMvnResolver(outStream).mavenResolver(library.groupId, library.artifactId,
-                                    library.version, projectRoot, false);
+            File tomlFile = new File(Paths.get(projectRoot.toString(), BALLERINA_TOML).toString());
```
Comment (hevayo) [inline misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/command/BindingsGenerator.java:149]: Here you can get the toml from the project without separately loading it project.currentPackage().ballerinaToml()

**Review by hevayo — CHANGES_REQUESTED** [review]

File: misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/command/BindingsGenerator.java (hunk 1 above)
Comment (IrushiL) [reply misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/command/BindingsGenerator.java]: Fixed

File: misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/utils/BindgenConstants.java [hunk 2]
```diff
@@ -43,12 +43,13 @@ private BindgenConstants() {
     public static final String CONSTANTS_FILE_NAME = "Constants.bal";
     public static final String ARRAY_BRACKETS = "[]";
     public static final String MODULES_DIR = "modules";
+    public static final String BALLERINA_TOML = "Ballerina.toml";
```
Comment (pramodya1994) [inline misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/utils/BindgenConstants.java:46]: Cant we use `ProjectConstants.BALLERINA_TOML` without redefining a new cont?

File: misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/utils/BindgenConstants.java [hunk 3]
```diff
@@ -43,12 +43,13 @@ private BindgenConstants() {
...
+    public static final String JAVA_11 = "java11";
```
Comment (pramodya1994) [inline misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/utils/BindgenConstants.java:47]: Cant we use `JvmTarget.JAVA_11.code()` without defining a const?

File: misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/utils/BindgenConstants.java (hunk 2 above)
Comment (IrushiL) [reply misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/utils/BindgenConstants.java]: Thanks, fixed it.

File: misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/utils/BindgenConstants.java (hunk 3 above)
Comment (IrushiL) [reply misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/utils/BindgenConstants.java]: Thanks, fixed it.

File: misc/ballerina-bindgen/src/test/java/org/ballerinalang/bindgen/BindgenCommandTest.java [hunk 4]
```diff
@@ -1,11 +1,174 @@
+/*
```
[the remark this answers is not in the document]
Comment (IrushiL) [reply misc/ballerina-bindgen/src/test/java/org/ballerinalang/bindgen/BindgenCommandTest.java:1]: Thanks, fixed it.

**Comment by mohanvive** [comment]
@pramodya1994 @IrushiL Shall we merge this, any reasons to keep this open?

**Review by pramodya1994 — APPROVED** [review]

**Comment by IrushiL** [comment]
> @pramodya1994 @IrushiL Shall we merge this, any reasons to keep this open?

Will get this merged soon.

File: misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/command/BindingsGenerator.java [hunk 5]
```diff
@@ -140,17 +139,30 @@ void generateJavaBindings() throws BindgenException {
...
-                    for (Library library : platformLibraries) {
-                        if (library.path != null) {
-                            classPaths.add(Paths.get(projectRoot.toString(), library.path).toString());
-                        } else if (library.groupId != null && library.artifactId != null && library.version != null) {
-                            new BindgenMvnResolver(outStream).mavenResolver(library.groupId, library.artifactId,
-                                    library.version, projectRoot, false);
+        if (env.getProjectRoot() != null) {
+            TomlDocument tomlDocument = env.getTomlDocument();
+            if (tomlDocument != null) {
+                PackageManifest packageManifest = ManifestBuilder.from(tomlDocument, null,
```
Comment (hevayo) [inline misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/command/BindingsGenerator.java:145]: Shall we refactor the nested if and for we can split it in to multiple functions

**Review by hevayo — CHANGES_REQUESTED** [review]

File: misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/command/BindingsGenerator.java (hunk 5 above)
Comment (IrushiL) [reply misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/command/BindingsGenerator.java]: Done

**Review by hevayo — APPROVED** [review]

**Pull request merged**